entry_types/scrolled/package/src/frontend/MediaPlayerControls.js

```diff
@@ -16,36 +16,61 @@ export function MediaPlayerControls(props) {
-                    inactive={playerState.userIdle &&
-                              (!focusOutlineVisible || !playerState.focusInsideControls) &&
-                              !playerState.userHoveringControls}
+                              onPlayButtonClick={playerActions.playBlessed}
+                              onPauseButtonClick={playerActions.pause}
+                              scrubTo={playerActions.scrubTo}
+                              seekTo={playerActions.seekTo}
+                              
+                              textTracksMenuItems={getTextTracksMenuItems(textTracks, t)}
+                              onTextTracksMenuItemClick={textTracks.select} />
+    );
+  }
+  else {
+    return (
+      <ClassicPlayerControls inset={props.configuration.position === 'full' || props.configuration.caption}
+                      type={props.type}
+                      currentTime={playerState.scrubbingAt !== undefined ?
+                              playerState.scrubbingAt : playerState.currentTime}
+                      bufferedEnd={playerState.bufferedEnd}
+                      duration={playerState.duration}
+
+                      isPlaying={playerState.isPlaying}
+                      inactive={playerState.userIdle &&
+                                (!focusOutlineVisible || !playerState.focusInsideControls) &&
+                                !playerState.userHoveringControls}
 
-                    onFocus={playerActions.focusEnteredControls}
-                    onBlur={playerActions.focusLeftControls}
-                    onMouseEnter={playerActions.controlsEntered}
-                    onMouseLeave={playerActions.controlsLeft}
+                      onFocus={playerActions.focusEnteredControls}
+                      onBlur={playerActions.focusLeftControls}
+                      onMouseEnter={playerActions.controlsEntered}
+                      onMouseLeave={playerActions.controlsLeft}
 
-                    play={playerActions.playBlessed}
-                    pause={playerActions.pause}
-                    scrubTo={playerActions.scrubTo}
-                    seekTo={playerActions.seekTo}
+                      play={playerActions.playBlessed}
+                      pause={playerActions.pause}
+                      scrubTo={playerActions.scrubTo}
+                      seekTo={playerActions.seekTo}
 
-                    textTracksMenuItems={getTextTracksMenuItems(textTracks, t)}
-                    onTextTracksMenuItemClick={textTracks.select}
+                      textTracksMenuItems={getTextTracksMenuItems(textTracks, t)}
+                      onTextTracksMenuItemClick={textTracks.select}
 
-                    qualityMenuItems={props.qualityMenuItems}
-                    onQualityMenuItemClick={props.onQualityMenuItemClick} />
-  )
+                      qualityMenuItems={props.qualityMenuItems}
+                      onQualityMenuItemClick={props.onQualityMenuItemClick} />
+    )
+  }
```

Comment:
There was a misunderstanding. I did not mean to suggest moving the `playerState`/`playerActions` props from `MediaPlayerControls` to `PlayerControls` since that:

* Does not fix the mentioned duplication - it has just been moved

* Does not allow simplifying the duplicated specs

* Does not fix [the storybook](https://github.com/codevise/pageflow/blob/master/entry_types/scrolled/package/src/frontend/__stories__/playerControls-stories.js#L28) which still expects to be able to use low level props like `duration` on `PlayerControls`.

* Moreover, the divide of responsibilities between `PlayerControls` and `MediaPlayerControls` is now gone: Before `PlayerControls` was about the low level mechanics and look of the player controls, `MediaPlayerControls` was about binding the player controls to player state.

What I meant was:

* Basically keep the props of `PlayerControls` the way they are on `master` and only add a `variant` prop.

* Align the prop names of `WaveformPlayerControls` with `ClassicPlayerControls` (e.g. `onPlayerButtonClick` vs `play`).

* Based on `variant` render either `WaveformPlayerControls` or `ClassicPlayerControls` in `PlayerControls` passing exactly the same props. This removes the need to have two long, subtly different prop lists. Not all props are used by both components, but that does not matter.

* `MediaPlayerControls` can then stay basically the way it is on `master` and set the `variant` based on `configuration.playerControlsVariant`.d

* The specs which are now duplicated for `WaveformPlayerControls` and `ClassicPlayerControls` can be put into a loop that iterates over the two possible variants and render the `PlayerControls` component passing the `variant`.

* This fixes the storybook since the interface of `PlayerControls` was only extended.